site/content/en/docs/api_sdk/sdk/highlevel-api.md

```diff
@@ -44,20 +48,177 @@ with make_client(host="http://localhost") as client:
+- reach to lower-level APIs with the corresponding members
+
+An instance of `Client` can be created directly by calling the class constructor
+or with the utility function `cvat_sdk.core.client.make_client()` which can handle
+some configuration for you. A `Client` can be configured with
+the `cvat_sdk.core.client.Config` class instance. A `Config` object can be passed to
+the `Client` constructor and then it will be available in the `Client.config` field.
+
+The `Client` class implements the [context manager protocol](https://docs.python.org/3/reference/datamodel.html#context-managers).
+When the context is closed, the session is finished, and the user is logged out
+automatically. Otherwise, these actions can be done with the `close()` and `logout()` methods.
+
+You can create and start using a `Client` instance this way:
+
+```python
+from cvat_sdk import make_client
+
+with make_client('localhost', port='8080', credentials=('user', 'password')) as client:
+    ...
+```
+
+The `make_client()` function handles configuration and object creation for you.
+It also allows to authenticate right after the object is created.
+
+If you need to configure `Client` parameters, you can do this:
+
+```python
+from cvat_sdk import Config, Client
+
+config = Config()
+# set up some config fields ...
+
+with Client('localhost:8080', config=config) as client:
+    client.login(('user', 'password'))
+    ...
+```
+
+You can specify server address both with and without the scheme. If the scheme is omitted,
+it will be deduced automatically.
+
+> The checks are performed in the following
+order: `https` (with the default port 8080), `http` (with the default port 80).
+In some cases it may lead to incorrect results - eg. you have 2 servers running on the
+same host at default ports. In such cases just specify the schema manually: `https://localhost`.
+
+When the server is located, its version is checked. If an unsupported version is found,
+an error can be raised or suppressed (controlled by `config.allow_unsupported_server`).
+If the error is suppressed, some SDK functions may not work as expected with this server.
+By default, a warning is raised and the error is suppressed.
+
+> Please note that all `Client` operations rely on the server API and depend on the current user
+rights. This affects the set of available APIs, objects and actions. For example, a regular user
+can only see and modify their tasks and jobs, while an admin user can see all the tasks etc.
+
+## Entities and Repositories
+
+_Entities_ represent objects on the server. They provide read access to object fields
+and implement additional relevant operations, including both the general Read-Update-Delete and
+object-specific ones. The set of available general operations depends on the object type.
+
+_Repositories_ provide management operations for corresponding _Entities_. Typically, you don't
+need to create _Repository_ objects manually. To obtain a _Repository_ object, use the
```

Comment:
Do you _ever_ need to create Repository objects manually? Seems like the "typically" could be removed.

Reply:
Currently the answer is no, it's never needed. But it wasn't done on purpose.